src/makevoice/MakeVoice.cpp

```diff
@@ -104,9 +179,64 @@ int wmain(int argc, __in_ecount(argc) WCHAR* argv[])
     if (SUCCEEDED(hr))
     {
         copyFile(
-            getSiblingFilePath("Vocalizer.exe").c_str(),
-            AUTOMATION_VOICE_HOME "\\Vocalizer.exe"
+            getSiblingFilePath(_T("Vocalizer.exe")).c_str(),
+            _T(AUTOMATION_VOICE_HOME "\\Vocalizer.exe")
         );
+
+        if (tputenv_s(_T("WORDS"), UTTERANCE_FOR_SETTING_DEFAULT_VOICE) != 0)
+        {
+            hr = E_FAIL;
+        }
+    }
+
+    if (SUCCEEDED(hr))
+    {
+        hr = runSubprocess(tstring(_T(AUTOMATION_VOICE_HOME "\\Vocalizer.exe")));
+    }
+
+    return hr;
+}
+
+HRESULT uninstall()
+{
+    HRESULT hr = updateDllRegistry(_T("AutomationTtsEngine.dll"), DllAction::uninstall);
+
+    if (!SUCCEEDED(hr))
+    {
+        return hr;
+    }
+
+    WindowsRegistry::RegistryPathParts pathParts;
+
+    hr = WindowsRegistry::splitRegistryPath(SPCAT_VOICES, &pathParts);
+
+    if (!SUCCEEDED(hr))
+    {
+        return hr;
+    }
+
+    bool result = WindowsRegistry::deleteNode(
+        pathParts.root,
+        pathParts.rest + _T("\\") + _T(AUTOMATION_VOICE_ID)
+    );
+
+    if (!result)
+    {
+        return E_FAIL;
+    }
+
+    return system("rmdir /Q /S \"" AUTOMATION_VOICE_HOME "\"") == 0 ? S_OK : E_FAIL;
+}
+
+int wmain(int argc, __in_ecount(argc) WCHAR* argv[])
+{
+    bool shouldInstall = parseArgs(argc, argv);
```

Comment:
What do you think of returning the `DllAction` value instead of the boolean? Then compare it to determine the function to call?